### PR TITLE
Show custom error screen for drug stock web view when there is no internet connection

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/2980/show-custom-error-screen-for-drug-stock-web-view-when-there-is-no-internet-connection

### CODE REVIEW

**Comment by msasikanth** [comment]
Since Janhavi already has a bunch of PRs to review, adding @vinaysshenoy as a reviewer.

File: app/src/main/assets/drug_stock_no_connection.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<html>
```
Comment (vinaysshenoy) [inline app/src/main/assets/drug_stock_no_connection.html:1]: Can we format this HTML, please?

File: app/src/main/assets/drug_stock_no_connection.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+<body style="background: #F0F2F5; font-family: Roboto, sans-serif; text-align: center;">
+<div style="font-size: 24px; font-weight: bold; padding: 24px 32px 8px 32px; color: #2F363D;">Not available</div>
+<div style="padding: 0 32px 24px 32px; color: #6C737A;">An internet connection is needed to use this feature.</div>
```
Comment (vinaysshenoy) [inline app/src/main/assets/drug_stock_no_connection.html:4]: These user visible strings need to be localised, no?

File: app/src/main/java/org/simple/clinic/home/report/ReportsWebViewClient.kt [hunk 3]
```diff
@@ -11,10 +15,29 @@ class ReportsWebViewClient(
   override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
     val url = request?.url?.toString() ?: return super.shouldOverrideUrlLoading(view, request)
 
+    Timber.d(url)
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/report/ReportsWebViewClient.kt:18]: Leftover log?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/main/assets/drug_stock_no_connection.html (hunk 1 above)
Comment (msasikanth) [reply app/src/main/assets/drug_stock_no_connection.html:1]: I did, Android Studio and Idea is formatting like this 🤷🏾

File: app/src/main/assets/drug_stock_no_connection.html (hunk 2 above)
Comment (msasikanth) [reply app/src/main/assets/drug_stock_no_connection.html:4]: Ideally yes, but I am unsure how translations would work for HTML files. Is there a specific format we should follow here? 🤔

File: app/src/main/assets/drug_stock_no_connection.html (hunk 1 above)
Comment (msasikanth) [reply app/src/main/assets/drug_stock_no_connection.html:1]: I will manually format the file and push the changes.

File: app/src/main/assets/drug_stock_no_connection.html (hunk 2 above)
Comment (msasikanth) [reply app/src/main/assets/drug_stock_no_connection.html]: Yeah. Let's go with a later option, we can add it now and improve it later if someone complains regarding the translations. For now, doing it during runtime seems like a bit of work to implement 😅

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**